src/makefun/main.py

```diff
@@ -177,10 +194,25 @@ def create_function(func_signature,             # type: Union[str, Signature]
     # name defaults
     user_provided_name = True
     if func_name is None:
-        # allow None for now, we'll raise a ValueError later if needed
+        # allow None, this will result in a lambda function being created
         func_name = getattr_partial_aware(func_impl, '__name__', None)
         user_provided_name = False
 
+    # co_name default
+    user_provided_co_name = True
+    if co_name is None:
+        if func_name is None:
+            co_name = '<lambda>'
+        else:
+            co_name = func_name
+        user_provided_co_name = False
```

Comment:
Probably more readable:

```suggestion
    user_provided_co_name = co_name is not None
    if not user_provided_co_name :
        if func_name is None:
            co_name = '<lambda>'
        else:
            co_name = func_name
```